### ISSUE TITLE
[invokers] Can we add interesttarget support to more elements

### ISSUE THREAD

**Issue by lukewarlow** [openui/open-ui#839]
See https://github.com/keithamus/invoker-buttons-proposal/issues/14 for more details and that repo in general for full details.

TLDR: `interesttarget` attribute currently proposed to be supported on buttons and links. Could we add support to `<area>`, `<input>`, `<textarea>`, `<select>`, `<summary>`. (`selectlist` in future would be on this list) along with potentially other elements?

Are there extra elements we can/should add this too?

**Comment by lukewarlow** [comment]
To clarify is there a specific reason against `<area>`?

**Comment by lukewarlow** [comment]
See https://github.com/openui/open-ui/issues/857 for a request for interest support on abbr elements.

**Comment by lukewarlow** [comment]
Just while I remember I thought I'd not it down one use case for interesttarget (at least the focus aspect) on inputs in general is the `<input list="foo">` type scenario where you want to be able to trigger a popover of suggestions while focused on an input. Though potentially this is better off being a platform editable combobox type setup and we don't expose the capability beyond that at least initially.

Much the same way that selectlist will have focusgroup behaviour (afaik) but we don't expose focusgroup yet.

**Comment by adactio** [comment]
Could `interestTarget` be used to provide a declarative option for some JavaScript APIs? I'm thinking specifcally of JavaScript APIs that require a user interaction (and so most demos involve the creation of a button element anyway) and that don't currently have an equivalent HTML element.

For example:

* The Fullscreen API
* The Geolocation API
* The Web Share API

For that last one, the default behaviour could be sharing the current URL. I've been [collecting use cases](https://github.com/adactio/share-button-type/issues/1) to inform [a proposal for `button type="share"`](https://github.com/adactio/share-button-type/issues/1) but `button interestarget="share"` would achieve the same goal.

Likewise, `button interesttartget="requestfullscreen"` and `button interesttarget="exitfullscreen` could be directly equivalent to `document.requestfullscreen` and `document.exitfullscreen` but without the need to create a button with JavaScript (or write any JavaScript at all).

I'm not sure how the `button interesttarget="geolocation"` would work other than perhaps populating an `input` element using `invoketarget`.

I realise that this issue was asking specifically for HTML elements, not JavaScript APIs, so I apologize if this is off-topic.

**Comment by lukewarlow** [comment]
@adactio could you open a new issue for that discussion?

Fwiw it would be invoketarget rather than interesttarget (click vs hover), also fullscreen API is already included! Glad to get validation that others believe that would be useful!

**Comment by adactio** [comment]
@lukewarlow Done!

<https://github.com/openui/open-ui/issues/972>

I didn't realise that the fullscreen API was already included—that bodes very well!

**Comment by yisibl** [comment]
`<input type=range>` absolutely need it! Typically used to display a tooltip.

![image](https://github.com/openui/open-ui/assets/2784308/010807e2-b17a-4049-babd-212ed83d7958)

Demo: https://codepen.io/yisi/pen/jOJWLve

**Comment by lukewarlow** [comment]
Another thought is something like an input type file where it is at its core a button. We support buttons should we allow hovering and focus triggering for a file input? It'd almost be good if there was some way to target the file-selector-button part but obviously can't do that with attributes.

**Comment by Westbrook** [comment]
When this gets revisited, ensuring we can support `#elementinternal.role="button"` or event just `role="button"` content would be amazing!!

If I didn't think it would take forever, I'd say this was a great place for the browser to start shipping element behaviors that custom element developers could leverage to be a more _natural_ part of the larger web platform...

**Comment by hunterloftis** [comment]
Yes, instead of trying to predetermine the canonical list of things that may sometimes behave similarly to buttons, being able to provide a role for interoperability between things like `<x-button>` and `popovertarget` would be elegant & powerful.

**Comment by keithamus** [comment]
> When this gets revisited, ensuring we can support `#elementinternal.role="button"` or event just `role="button"` content would be amazing!!

I believe the HTML spec takes an architectural hard line that aria roles and attributes should not impact HTML features, and should only be used to map to accessibility information.

**Comment by scottaohara** [comment]
correct @keithamus.  ARIA is not meant to have any functional or styling impact on HTML at all.  merely a way to change the way an element is exposed to the a11y tree.  

that said, i do think it'd make sense for this to work on custom elements that have been properly setup via element internals.  but `div role=button` is still just a div at its core.

**Comment by chrisdholt** [comment]
> correct @keithamus. ARIA is not meant to have any functional or styling impact on HTML at all. merely a way to change the way an element is exposed to the a11y tree.
> 
> that said, i do think it'd make sense for this to work on custom elements that have been properly setup via element internals. but `div role=button` is still just a div at its core.

Huge +1 for custom element support via ElementInternals.

**Comment by YummyBacon5** [comment]
I think that `tabindex=0` should also be supported since I've seen sites have tooltips on elements which aren't buttons.

Or if not, then `abbr` could have it supported instead. Since this element is meant to use the `title` attribute for its expanded abbreviation, however, `title` isn't accessible to most browsers.

**Comment by SilentImp** [comment]
Hi, 
I want to add additional problems and use cases to consider.
Invokers can be added to buttons/inputs—to interactive elements—to avoid the "pit of success." It makes sense. 

However, there is an issue related to that: inputs and buttons are replaceable elements. As a result, there is no way you can make them behave as normal text in the flow. Applying hyphens, flow through lines and all the rest. 

Yes, it's the presentational problem and it can be solved from the CSS side:
- [[css-forms] Allow making <button> a real inline (or block/flex/etc)](https://github.com/w3c/csswg-drafts/issues/3226) _Feb 16, 2021_
- [[css-ui] appearance: base to enable interoperable styling of controls/components](https://github.com/w3c/csswg-drafts/issues/5998) _Oct 19, 2018_

At the same time maybe we can solve it from invokers side as well. Could it be faster, more efficient or better from any other point of view?

I see a lot of typography-related use cases for popovers/invokers. 
**For example:** you have a paragraph of text. In the word are abbreviations. You decide to move away from the `abbr` attribute `title` to have a fancy-looking and more accessible popover with full form. You have to write the abbreviation on the button ... and now it's not behaving the same way as before. Line height and baseline can give you a headache. 

But ok, abbreviations are not long. But what about reference text?
I have a paragraph of news text. There are references to some past events, which are not the subject of this article. I want to click on this event reference text and show the cut-in side notes to the left or right of the column. I wrapped reference text in a button, but now it doesn't properly flow into the next line. 

Of course, there can be UI workarounds: add a button info after text and make it super, or something. But it seems hacky without a reason. 

What are your thoughts on this matter?
